File: rust/src/models/update_coupon_request.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::{coupon_discount::CouponDiscount, plan_id::PlanId};

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct UpdateCouponRequest {
    #[serde(skip_serializing_if = "Option::is_none")]
    pub description: Option<String>,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub discount: Option<CouponDiscount>,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub plan_ids: Option<Vec<PlanId>>,
}

impl UpdateCouponRequest {
    pub fn new() -> Self {
        Self {
            description: None,
            discount: None,
            plan_ids: None,
        }
    }
}

File: typescript/src/models/updateCouponRequest.ts
// this file is @generated
import { type CouponDiscount, CouponDiscountSerializer } from "./couponDiscount";
import { type PlanId, PlanIdSerializer } from "./planId";

export interface UpdateCouponRequest {
  description?: string | null;

  discount?: CouponDiscount | null;

  planIds?: PlanId[] | null;
}

export const UpdateCouponRequestSerializer = {
  _fromJsonObject(object: any): UpdateCouponRequest {
    return {
      description: object["description"],
      discount:
        object["discount"] != null
          ? CouponDiscountSerializer._fromJsonObject(object["discount"])
          : undefined,
      planIds:
        object["plan_ids"] != null
          ? object["plan_ids"].map((item: any) => PlanIdSerializer._fromJsonObject(item))
          : undefined,
    };
  },

  _toJsonObject(self: UpdateCouponRequest): any {
    return {
      description: self.description,
      discount:
        self.discount != null
          ? CouponDiscountSerializer._toJsonObject(self.discount)
          : undefined,
      plan_ids:
        self.planIds != null
          ? self.planIds.map((item: any) => PlanIdSerializer._toJsonObject(item))
          : undefined,
    };
  },
};
